src/elli_http.erl

```diff
@@ -668,7 +674,13 @@ connection_token(#req{version = {1, 0}, headers = Headers}) ->
 connection_token(#req{version = {0, 9}}) ->
     <<"close">>.
 
-
+%% @doc Return the preferred session handling setting to close or keep the
+%% current session alive based on the presence of a header or the standard
+%% default based on the version of HTTP of the request.
+-spec close_or_keepalive(Req, Headers) -> KeepaliveOpt when
+      Req       :: elli:req(),
+      Headers   :: elli:headers(),
+      KeepaliveOpt :: close | keep_alive.
 close_or_keepalive(Req, UserHeaders) ->
     case get_header(?CONNECTION_HEADER, UserHeaders) of
```

Comment:
Not really related to this PR, but why don't we do this instead, I wonder?

```erlang
close_or_keepalive(Req, UserHeaders) ->
    case get_header(?CONNECTION_HEADER, UserHeaders, connection_token(Req)) of
        <<"Keep-Alive">> -> keep_alive;
        <<"close">> -> close
    end.

Reply:
Good point.

Reply:
I have this change pushed up as well, but for some reason I'm not getting a failure in the SSL test suite I'm not able to reproduce locally.

I had hoped it was explained by a difference in the version of Erlang I was running locally (was 22.3.4.12) but installing a couple of different versions, including 22.3.4.2 I still can't reproduce.

Any suggestions, or should I revert this last change for now?

Reply:
I'd say revert it for now. No need to let an attempted drive-by optimisation impede this PR.

Reply:
I guess in the "optimised" code `connection_token/1` is always evaluated, whereas before if `get_header/2` didn't return `undefined` then `connection_token/1` is never called. Seems very fishy, but apparently we need the delayed evaluation on 22.3.4.2...

Reply:
So I reverted that last change and I'm still getting the same error (`{badmatch, []}` in elli_ssl_tests), this time on the OTP 21.3.8.16 release, so it looks like it may be a timing issue unrelated to this change. Gonna try and push that change up again just to see if it passes but I can always revert again if preferred.

Reply:
Sounds good to me. Would you mind opening an issue mentioning the timing issue?